cli/cmd/cluster_kubeconfig.go

```diff
@@ -87,21 +86,27 @@ func (r *runners) kubeconfigCluster(_ *cobra.Command, args []string) error {
 				return errors.Wrap(err, "create kubeconfig dir")
 			}
 		}
-		if err := ioutil.WriteFile(r.args.kubeconfigPath, kubeconfig, 0644); err != nil {
+		if err := os.WriteFile(r.args.kubeconfigPath, kubeconfig, 0644); err != nil {
 			return errors.Wrap(err, "write kubeconfig")
 		}
 
 		fmt.Printf("kubeconfig written to %s\n", r.args.kubeconfigPath)
 		return nil
 	}
 
-	tmpFile, err := ioutil.TempFile("", "replicated-kubeconfig")
+	tmpFile, err := os.CreateTemp("", "replicated-kubeconfig")
 	if err != nil {
 		return errors.Wrap(err, "create temp file")
 	}
-	defer os.Remove(tmpFile.Name())
-	if err := ioutil.WriteFile(tmpFile.Name(), kubeconfig, 0644); err != nil {
-		return errors.Wrap(err, "write temp file")
+	defer func() {
+		tmpFile.Close()
+		os.Remove(tmpFile.Name())
+	}()
+	if _, err := tmpFile.Write(kubeconfig); err != nil {
```

Comment:
There is a race condition.  This file is read before the buffer contents are flushed to disk.  That's only guaranteed to work when the same file handle is used, but not if file is opened a second time.